### PR TITLE
[googletts] Replace custom TTS cache with common TTS cache

### PR DESCRIPTION
**Pull request by dalgwen**
Uses the common TTS cache instead of a custom one.
No new features

Relates to openhab/openhab-addons#15110

Advantages 
- use common configuration
- cache size configuration available
- better concurrency management
- simplifying googletts code.

### CODE REVIEW

**Comment by dalgwen** [comment]
I didn't update the internationalization files for langage other than default.

I believe it should be done inside the dedicated tool ? If not, please tell me.

File: bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java [hunk 1]
```diff
@@ -361,4 +347,19 @@ private AudioFormat parseAudioFormat(byte[] audio) throws TTSException {
...
+            @NonNull AudioFormat requestedFormat) {
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            byte[] bytesOfMessage = (config.toConfigString() + text + requestedFormat).getBytes(StandardCharsets.UTF_8);
+            String hash = String.format("%032x", new BigInteger(1, md.digest(bytesOfMessage)));
+            return ((GoogleTTSVoice) voice).getTechnicalName() + "_" + hash;
+        } catch (NoSuchAlgorithmException e) {
+            // should not happen
+            logger.error("Could not create MD5 hash for '{}'", text, e);
+            return "nomd5algorithm";
```
Comment (lolodomo) [inline bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java:362]: A WARN level should be better.
I am asking myself if you could not just call the super class if your method fails ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

**Comment by lolodomo** [comment]
This TTS service is using a ByteArrayAudioStream. So we can merge this PR without waiting any enhancement of audio sinks.

File: bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java:362]: I already read your comment in the other PR, so forget my remark about calling the super class.

Please just change the log level.

Comment (lolodomo) [reply bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java:362]: @dalgwen : please consider changing the log level and then I will merge.

Comment (dalgwen) [reply bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java]: Done

**Review by lolodomo — APPROVED** [review]
LGTM thank you

**Pull request merged**